src/mds/MDCache.cc

```diff
@@ -3083,6 +3083,7 @@ void MDCache::handle_resolve(MMDSResolve *m)
 
   if (mds->get_state() < MDSMap::STATE_RESOLVE) {
     if (mds->get_want_state() == CEPH_MDS_STATE_RESOLVE) {
+      discard_delayed_resolve(from);
```

Comment:
Right, but process_delayed_resolve will only call through to handle_resolve if there is something in MDCache::delayed_resolve, and something only gets put in there once we are >=STATE_RESOLVE.  So as long as we are <STATE_RESOLVE, delayed_resolve is guaranteed to be empty